components/brave_wallet/browser/solana_provider_impl.h

```diff
@@ -42,7 +50,40 @@ class SolanaProviderImpl final : public mojom::SolanaProvider {
   void Request(base::Value arg, RequestCallback callback) override;
 
  private:
+  void OnConnect(ConnectCallback callback,
+                 const absl::optional<std::string>& account,
+                 mojom::SolanaProviderError error,
+                 const std::string& error_message);
+  void OnEagerlyConnect(ConnectCallback callback,
+                        const absl::optional<std::string>& account,
+                        bool allowed);
+
+  // mojom::KeyringServiceObserver
+  void KeyringCreated(const std::string& keyring_id) override {}
+  void KeyringRestored(const std::string& keyring_id) override {}
+  void KeyringReset() override {}
+  void Locked() override {}
+  void Unlocked() override {}
+  void BackedUp() override {}
+  void AccountsChanged() override {}
+  void AutoLockMinutesChanged() override {}
+  void SelectedAccountChanged(mojom::CoinType coin) override;
+
+  // This map is used to maintain connected status for each frame, it is a
+  // separate non persistent status from site permission. It depends on if a
+  // site successfully call connect or not, calling disconnect will remove
+  // itself from this map. Note that site permission is required for a site to
+  // do connect, if an user reject the connect request, connect would fail. On
+  // the other hand, if the user approve the connect request, site permission
+  // will be saved and future connect from the same site will not ask user for
+  // permission again until the permission is removed.
+  base::flat_set<std::string> connected_map_;
```

Comment:
nit: connected_set

Reply:
also s/map/set in the above comment.

Reply:
fixed in 46a8acf771da14616feee025e6f78c9cffced934